DOCS-422 update Cluster-configuration-settings.md (#85)
aerospike/kubernetes-operator: docs/Cluster-configuration-settings.md

description: Cluster Configuration Settings
---

> Aerospike cluster configuration settings are found in the Aerospike cluster Custom Resource (CR). You can edit this file to make changes to your Aerospike cluster.

Reviewer: ```suggestion
Aerospike cluster configuration settings are in the Aerospike cluster Custom Resource (CR). You can edit this file to make changes to your Aerospike cluster.
```

> The Operator Custom Resource Definition (CRD) specifies the CR the Operator uses to manage the cluster.

Reviewer 2: Hmm another "why aren't we linking out" question.

## Storage

The storage section configures persistent volume devices to provision and attach to the containers.

> This section is required by default for persisting the Aerospike work directory. The work directory should be stored on persistent storage to ensure pod restarts do not reset Aerospike server metadata files.

Reviewer: ```suggestion
This section is required by default for persisting the Aerospike work directory. Store the work directory on persistent storage to ensure pod restarts do not reset Aerospike server metadata files.
```

For filesystem volumes, initMethod can be `none` or `deleteFiles`.
For block volumes, initMethod can be `none`, `dd` or `blkdiscard`.

:::note

> `blkdiscard` will only work for devices which support [TRIM](https://en.wikipedia.org/wiki/Trim_%28computing%29). For AWS please refer to the [storage volumes guide](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html#instance-store-volumes) to check TRIM support. If TRIM is not supported please use the slower `dd` if your devices need initialization. For other devices please verify the support for TRIM command.

Reviewer: ```suggestion
`blkdiscard` only works for devices which support [TRIM](https://en.wikipedia.org/wiki/Trim_%28computing%29). For AWS please refer to the [storage volumes guide](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html#instance-store-volumes) to check TRIM support. If TRIM is not supported please use the slower `dd` to initialize your devices. For other devices please verify the support for TRIM command.
```

For filesystem volumes, initMethod can be `none` or `deleteFiles`.
For block volumes, initMethod can be `none`, `dd` or `blkdiscard`.

:::note

> `blkdiscard` will only work for devices that support [TRIM](https://en.wikipedia.org/wiki/Trim_%28computing%29). For AWS please refer to the [storage volumes guide](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html#instance-store-volumes) to check TRIM support. If TRIM is not supported please use the slower `dd` if your devices need initialization. For other devices please verify the support for TRIM command.

Reviewer: ```suggestion
`blkdiscard` only works for devices that support [TRIM](https://en.wikipedia.org/wiki/Trim_%28computing%29). For AWS please refer to the [storage volumes guide](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html#instance-store-volumes) to check TRIM support. If TRIM is not supported please use the slower `dd` to initialize your devices. For other devices please verify the support for TRIM command.
```

```

#### ConfigMap

> A Kubernetes ConfigMap can be mounted as  a volume. ConfigMaps hold non-confidential data like configuration for applications running in your custom init containers or sidecars.

Reviewer: ```suggestion
A Kubernetes ConfigMap can be mounted as a volume. ConfigMaps hold non-confidential data like configuration for applications running in your custom init containers or sidecars.
```

## Aerospike Config

The YAML form of Aerospike server configuration. See [Aerospike Configuration](Aerospike-configuration-mapping.md) for details.

## Rack Config

> Configures the operator to deploy a rack-aware Aerospike cluster. Pods will be deployed in given racks based on the given configuration.

Reviewer: ```suggestion
Configures the operator to deploy a rack-aware Aerospike cluster. Pods are deployed in given racks based on the given configuration.
```
